napari/components/dims.py

```diff
@@ -110,6 +125,26 @@ def mode(self):
         """List of DimsMode: List of DimsMode, one for each dimension."""
         return copy(self._mode)
 
+    @property
+    def axis_labels(self):
```

Comment:
The getter-setter approach here makes it difficult for me to allow code such as `my_dims_instance.axis_labels[3] = 'time'`. Not sure how to resolve it, perhaps making a class for it will help.

Reply:
yes, we have that issue with all the properties where we use the setter approach, so that issue is definitely bigger than this PR.

unfortunately directly updating a list like that doesn't trigger the event emission that we need. it's definitely not ideal, but i'm not sure how to fix it